objects.js

```diff
@@ -4875,6 +4894,41 @@ StageMorph.prototype.penTrailsMorph = function () {
     return morph;
 };
 
+StageMorph.prototype.addLayer = function(canvas, name) {
+    var layer = {
+        canvas: canvas,
+        name: name };
+
+    if (!this.getLayer(name)) {
+        this.layers.push(layer)
+    }
+
+    return layer;
+};
+
+StageMorph.prototype.getLayer = function(name) {
```

Comment:
is there any particular reason for not using ```detect()``` here (and also further down)? I know that we've recently re-"introduced" the old imperative ```for``` loop to Morphic, but that was really just a performance optimization for two (I think) very special cases in which *every* Morph in the whole world gets enumerated at every mouse event. In these cases here I'm not assuming there to be hundreds or thousands of layers, so I'd suggest just using HOFs as we usually do :-)

(no need to edit or change anything just now, @bromagosa , I only wanted to log this reminder to myself here)!